crates/sui-protocol-config/src/lib.rs

```diff
@@ -1286,10 +1286,16 @@ pub struct ProtocolConfig {
     /// Transactions will be cancelled after this many rounds.
     max_deferral_rounds_for_congestion_control: Option<u64>,
 
-    /// If >0, congestion control will allow up to one transaction per object to exceed
-    /// the configured maximum accumulated cost by the given amount.
+    /// If >0, congestion control will allow the configured maximum accumulated cost per object
+    /// to be exceeded by at most the given amount. Only one limit-exceeding transaction per
+    /// object will be allowed, unless bursting is configured below.
     max_txn_cost_overage_per_object_in_commit: Option<u64>,
 
+    /// If >0, congestion control will allow transactions in total cost equaling the
+    /// configured amount to exceed the configured maximum accumulated cost per object.
+    /// As above, up to one transaction per object exceeding the burst limit will be allowed.
+    allowed_txn_cost_overage_burst_per_object_in_commit: Option<u64>,
```

Comment:
when we discussed this last, we seemed to have settled on the mental model that the two params could be thought of as "the rate at which scheduling debt is paid down" and "the debt ceiling"

i think this model is much clearer, because i'm having trouble understanding your comments above even though I at least think I understand the system.

Reply:
it's slightly complicated by the requirement to maintain prior behavior when the new param is not set

 1. `max_accumulated_txn_cost_per_object_in_commit`
 - is always the rate at which debt is paid down
 - also the absolute limit, if it's the only thing set

2.  `max_txn_cost_overage_per_object_in_commit`
 if set with #1,
 - soft debt ceiling is #1
 - absolute limit is #1+#2
 (this is how it is currently configured to work, with #2 == u64 max)

3. `allowed_txn_cost_overage_burst_per_object_in_commit`
if set with #1 and #2 (must be <= #2 per assert)
- soft debt ceiling is #1+#3
- absolute limit is still #1+#2